docs/openid.md

```diff
@@ -47,6 +47,15 @@ setting in your configuration file (or uncomment one of the existing examples).
 See [sample_config.yaml](./sample_config.yaml) for some sample settings, as well as
 the text below for example configurations for specific providers.
 
+One thing to look out for if you are using a generic OpenID Connect provider is to
+manually set the config to use
+```yaml
+user_profile_method: userinfo_endpoint
+```
+as including the `openid` claim does NOT cause the default `auto` config to reliably
+use the userinfo endpoint, contrary to the homeserver.yaml comments. This may cause 
+the `preferred_username` to not be available when setting the localpart_template.
```

Comment:
This may be more a case of updating the sample config comments instead?

Reply:
Specifically, I think adding an explanation for something to look out for when setting `user_profile_method` would probably be better put in the config file which contains a description for each option. See https://github.com/matrix-org/synapse/blob/4bbd53545041c9295fbae5dd0cefdd66d55d0b53/synapse/config/oidc_config.py#L150-L155